xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlay.h

```diff
@@ -34,68 +35,34 @@ class CDVDOverlay
     iPTSStopTime = 0LL;
     bForced = false;
     replace = false;
-    m_references = 1;
     m_textureid = 0;
     m_enableTextAlign = false;
     m_overlayContainerFlushable = true;
     m_setForcedMargins = false;
   }
 
-  CDVDOverlay(const CDVDOverlay& src)
+  CDVDOverlay(const CDVDOverlay& src) : std::enable_shared_from_this<CDVDOverlay>(src)
   {
     m_type        = src.m_type;
     iPTSStartTime = src.iPTSStartTime;
     iPTSStopTime  = src.iPTSStopTime;
     bForced       = src.bForced;
-    replace       = src.replace;
-    m_references  = 1;
+    replace = src.replace;
     m_textureid = 0;
     m_enableTextAlign = src.m_enableTextAlign;
     m_overlayContainerFlushable = src.m_overlayContainerFlushable;
     m_setForcedMargins = src.m_setForcedMargins;
   }
 
-  virtual ~CDVDOverlay()
-  {
-    assert(m_references == 0);
-  }
-
-  /**
-  * increase the reference counter by one.
-  */
-  CDVDOverlay* Acquire()
-  {
-    m_references++;
-    return this;
-  }
-
-  /**
-  * decrease the reference counter by one.
-  */
-  int Release()
-  {
-    m_references--;
-    int ret = m_references;
-    if (m_references == 0)
-      delete this;
-    return ret;
-  }
-
-  /**
-   * static release function for use with shared ptr for example
-   */
-  static void Release(CDVDOverlay* ov)
-  {
-    ov->Release();
-  }
+  virtual ~CDVDOverlay() = default;
 
-  bool IsOverlayType(DVDOverlayType type) { return (m_type == type); }
+  bool IsOverlayType(DVDOverlayType type) const { return (m_type == type); }
 
   /**
    * return a copy to VideoPlayerSubtitle in order to have hw resources cleared
    * after rendering
    */
-  virtual CDVDOverlay* Clone() { return Acquire(); }
+  virtual std::shared_ptr<CDVDOverlay> Clone() { return shared_from_this(); }
```

Comment:
`shared_from_this` should be in the std namespace, right?

Reply:
Nope, this is a member function inherited from `std::enable_shared_from_this`.